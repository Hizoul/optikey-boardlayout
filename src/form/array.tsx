import { getMapToFromProps, IFieldProps, prependPrefix, SharedField, useArray } from "@xpfw/form"
import { observer } from "mobx-react-lite"
import * as React from "react"
import { Button, IconButton } from "@material-ui/core"
import DeleteIcon from "@material-ui/icons/Delete"
import AddIcon from "@material-ui/icons/Add"

const ArrayField: React.FunctionComponent<IFieldProps> = observer((props) => {
  let mapTo = getMapToFromProps(props)
  const arrayHelper = useArray(props.schema, mapTo, props.prefix)
  let hideArrayType = mapTo.indexOf("keyActions") !== -1 && arrayHelper.length < 2
  console.log("ARRAY ", hideArrayType, mapTo, mapTo.indexOf("keyActions") !== -1, arrayHelper.length)
  return (
    <div>
      {arrayHelper.fields.map((field) => {
        return (
        <div className={`flex flex1 center ${hideArrayType ? "" : "objectMargin"}`} key={prependPrefix(field.mapTo, field.prefix)}>
          <SharedField schema={field.schema} mapTo={field.mapTo} prefix={field.prefix} />
          {hideArrayType ? null : <IconButton
            onClick={field.decreaseSize}
            color="secondary"
          >
            <DeleteIcon />
          </IconButton>}
        </div>
      )})}
      {hideArrayType ? null : 
        <Button
          onClick={arrayHelper.increaseSize}
          variant="contained"
          fullWidth
          endIcon={<AddIcon />}
        >
          Add
        </Button>}
    </div >
  )
})

export default ArrayField
