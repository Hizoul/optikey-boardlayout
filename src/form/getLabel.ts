const getLabel = (label: string) => {
  let shortened = label
  const i = label.lastIndexOf("]")
  if (i !== -1) {
    shortened = shortened.substring(i + 2)
  }
  shortened = shortened.replace(/([A-Z])/g, ' $1')
  shortened = shortened.charAt(0).toUpperCase() + shortened.slice(1)
  return shortened
}

export default getLabel