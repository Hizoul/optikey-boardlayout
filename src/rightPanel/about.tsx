import { AppBar, IconButton, Link, Toolbar } from "@material-ui/core"
import ExpansionPanel from "@material-ui/core/ExpansionPanel"
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails"
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary"
import Slide from "@material-ui/core/Slide"
import Dialog from "@material-ui/core/Dialog"
import DialogContent from "@material-ui/core/DialogContent"
import Typography from "@material-ui/core/Typography"
import CloseIcon from "@material-ui/icons/Close"
import ExpandMoreIcon from "@material-ui/icons/ExpandMore"
import { observable } from "mobx"
import { observer } from "mobx-react-lite"
import * as React from "react"
import licenses from "../util/licenses"

const classes: any = {
  appBar: {
    position: "relative"
  },
  flex: {
    flex: 1
  }
}
const Transition  = (props: any) => {
  return <Slide direction="up" {...props} />
}

const aboutIsOpen = observable.box(false)
const handleCloseAbout = () => {
  aboutIsOpen.set(false)
}
const handleOpenAbout = () => aboutIsOpen.set(true)

const AboutPanel = observer(() => {
  return (
    <ExpansionPanel defaultExpanded>
      <ExpansionPanelSummary  expandIcon={<ExpandMoreIcon />}>
        <Typography>About</Typography>
      </ExpansionPanelSummary>
      <ExpansionPanelDetails>
        <Typography>
          Written with ❤️ by <Link href="https://github.com/hizoul">hizoul</Link>.<br/>
          Special thanks to&nbsp;
          <Link href="https://github.com/JuliusSweetland">Julius Sweetland</Link>
          &nbsp;for writing&nbsp;
          <Link href="https://github.com/OptiKey/OptiKey">
            OptiKey
          </Link> and&nbsp;
          <Link
            onClick={handleOpenAbout}
            href="#"
          >
            all authors of open source libraries
          </Link>
          &nbsp;enabling the existence of this tool.<br />
          Refer to the <Link href="https://github.com/OptiKey/OptiKey/wiki/Creating-and-Using-Dynamic-Keyboards">custom keyboard documentation</Link> for explanations on what the settings mean.
        </Typography>
      </ExpansionPanelDetails>
      <Dialog
        fullScreen
        open={aboutIsOpen.get()}
        onClose={handleCloseAbout}
        TransitionComponent={Transition}
      >
        <AppBar className={classes.appBar}>
          <Toolbar>
            <Typography variant="h6" color="inherit" className="flex1">
              Third Party Licenses
            </Typography>
            <IconButton color="inherit" onClick={handleCloseAbout} aria-label="Close">
              <CloseIcon />
            </IconButton>
          </Toolbar>
        </AppBar>
        <DialogContent>
          <Typography style={{margin: "5rem 0rem"}}>
            {licenses.split("\n").map((s) => <div key={s}>{s}</div>)}
          </Typography>
        </DialogContent>
      </Dialog>
    </ExpansionPanel>
  )
})

export default AboutPanel
export {Transition, handleCloseAbout, handleOpenAbout, aboutIsOpen}