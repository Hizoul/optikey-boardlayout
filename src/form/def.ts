import { ComponentRegistry, ExtendedJSONSchema } from "@xpfw/form"
import { registerComponents } from "@xpfw/form-web"
import BooleanField from "./boolean"
import SelectField from "./select"
import TextField from "./text"
import ObjectField from "./object"
import ArrayField from "./array"

registerComponents()
ComponentRegistry.registerComponent("number", TextField)
ComponentRegistry.registerComponent("string", TextField)
ComponentRegistry.registerComponent("boolean", BooleanField)
ComponentRegistry.registerComponent("object", ObjectField)
ComponentRegistry.registerComponent("array", ArrayField)
ComponentRegistry.registerComponent("string", SelectField, "select")
const actionNames = ["Action", "ChangeKeyboard", "KeyDown", "KeyUp", "Loop", "Text", "Wait"]
const keyActionTypes: any[] = actionNames.map((v) =>{return {value: v, label: v}})
const keyActionObjectSchema = {
  type: "object",
  title: "keyActionObject",
  properties: {
    actionType: {
      type: "string", theme: "select", selectOptions: keyActionTypes,
      description: "The action type determines what kind of input is sent to the OS on key press."}
  }
}

const nameSchema: ExtendedJSONSchema = {
  title: "Name", type: "string",
  description: "The name that will be shown in OptiKey"
}

const widthSchema: ExtendedJSONSchema = {
  title: "Width", type: "string",
  description: "Percent of screen. if negative then use the sum of the value and the screen's dimension"
}

const heightSchema: ExtendedJSONSchema = {
  title: "Height", type: "string",
  description: "Percent of screen. if negative then use the sum of the value and the screen's dimension"
}
const horizontalSchema: ExtendedJSONSchema = {
  title: "HorizontalOffset", type: "string",
  description: "Percent of screen. if negative then use the sum of the value and the screen's dimension"
}

const verticalSchema: ExtendedJSONSchema = {
  title: "VerticalOffset", type: "string",
  description: "Percent of screen. if negative then use the sum of the value and the screen's dimension"
}

const symbolSchema: ExtendedJSONSchema = {
  title: "SymbolMargin", type: "number",
  description: "Amount of empty space left for symbols"
}

const borderSchema: ExtendedJSONSchema = {
  title: "BorderThickness", type: "string",
  description: "Controls how thick the lines between individual keys are"
}

const rowsSchema: ExtendedJSONSchema = {
  title: "Grid.Rows", type: "number",
  description: "Controls the amount of keyboard rows"
}

const colsSchema: ExtendedJSONSchema = {
  title: "Grid.Cols", type: "number",
  description: "Controls the amount of keyboard columns"
}

const persistNewStateSchema: ExtendedJSONSchema = {
  title: "PersistNewState", type: "boolean",
  description: "Indiates whether the size and position of this keyboard will be saved"
}

const windowStateSchema: ExtendedJSONSchema = {
  title: "WindowState", type: "string", theme: "select",
  selectOptions: [
    {label: "Floating", value: "Floating"},
    {label: "Docked", value: "Docked"},
    {label: "Maximised", value: "Maximised"}],
  description: "Controls how the OptiKey window should behave"
}
const hideFromListSchema: ExtendedJSONSchema = {
  title: "HideFromKeyboardMenu", type: "boolean",
  description: "controls whether or not this keyboard will be listed in the dynamic keyboard selection menu. You may wish to set this equal to False for smaller utility-type keyboards."
}

const dockSizeSchema: ExtendedJSONSchema = {
  title: "DockSize", type: "string", theme: "select",
  selectOptions: [
    {label: "Full", value: "Full"},
    {label: "Collapsed", value: "Collapsed"}
  ],
  description: "Controls the size of the dock"
}

const positionSchema: ExtendedJSONSchema = {
  title: "Position", type: "string", theme: "select",
  description: "Controls where the Keyboard will be shown",
  selectOptions: [
    {label: "TopLeft", value: "TopLeft"},
    {label: "Top", value: "Top"},
    {label: "TopRight", value: "TopRight"},
    {label: "Left", value: "Left"},
    {label: "Right", value: "Right"},
    {label: "BottomLeft", value: "BottomLeft"},
    {label: "Bottom", value: "Bottom"},
    {label: "BottomRight", value: "BottomRight"}
  ],
}

const backgroundColorSchema: ExtendedJSONSchema = {
  title: "BackgroundColor", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const borderColorSchema: ExtendedJSONSchema = {
  title: "BorderColor", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const keyboardPrefix = "Keyboard"
const hoverKey = `hover`
const dragKey = `drag`
const activeKey = `active`

export {
  nameSchema, heightSchema, borderSchema, symbolSchema, rowsSchema, colsSchema, keyboardPrefix,
  activeKey, dragKey, hoverKey, persistNewStateSchema, windowStateSchema,
  dockSizeSchema, positionSchema, widthSchema, horizontalSchema, verticalSchema,
  backgroundColorSchema, borderColorSchema, hideFromListSchema, keyActionObjectSchema, actionNames
}
