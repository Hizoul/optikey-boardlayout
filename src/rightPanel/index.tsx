import ExpansionPanel from "@material-ui/core/ExpansionPanel"
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails"
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary"
import Typography from "@material-ui/core/Typography"
import ExpandMoreIcon from "@material-ui/icons/ExpandMore"
import { SharedField } from "@xpfw/form"
import * as React from "react"
import {
  borderSchema, colsSchema, heightSchema, keyboardPrefix,
  nameSchema, rowsSchema, symbolSchema, persistNewStateSchema, windowStateSchema,
  dockSizeSchema, positionSchema, widthSchema, horizontalSchema, verticalSchema,
  backgroundColorSchema, borderColorSchema, hideFromListSchema
} from "../form/def"
import AboutPanel from "./about"
import KeyboardImporter from "./import"
import KeyEditor from "./keyEditor"
import { resizeEventListener } from "../keyboard"

const resizeTriggerer = () => setTimeout(resizeEventListener, 1000)

const RightPanel: React.FunctionComponent<any> = () => {
  return (
    <div className="rightPanel">
      <KeyboardImporter />
      <ExpansionPanel defaultExpanded={false} onChange={resizeTriggerer}>
        <ExpansionPanelSummary  expandIcon={<ExpandMoreIcon />}>
          <Typography>Global Settings</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <div className="vertical flex1">
            <SharedField schema={nameSchema} prefix={keyboardPrefix} />
            <SharedField schema={rowsSchema} prefix={keyboardPrefix} />
            <SharedField schema={colsSchema} prefix={keyboardPrefix} />
            <SharedField schema={heightSchema} prefix={keyboardPrefix} />
            <SharedField schema={widthSchema} prefix={keyboardPrefix} />
            <SharedField schema={borderSchema} prefix={keyboardPrefix} />
            <SharedField schema={symbolSchema} prefix={keyboardPrefix} />
            <SharedField schema={positionSchema} prefix={keyboardPrefix} />
            <SharedField schema={persistNewStateSchema} prefix={keyboardPrefix} />
            <SharedField schema={windowStateSchema} prefix={keyboardPrefix} />
            <SharedField schema={dockSizeSchema} prefix={keyboardPrefix} />
            <SharedField schema={horizontalSchema} prefix={keyboardPrefix} />
            <SharedField schema={verticalSchema} prefix={keyboardPrefix} />
            <SharedField schema={backgroundColorSchema} prefix={keyboardPrefix} />
            <SharedField schema={borderColorSchema} prefix={keyboardPrefix} />
            <SharedField schema={hideFromListSchema} prefix={keyboardPrefix} />
          </div>
        </ExpansionPanelDetails>
      </ExpansionPanel>
      <KeyEditor />
      <AboutPanel />
    </div>
  )
}

export default RightPanel
export {
  resizeTriggerer
}