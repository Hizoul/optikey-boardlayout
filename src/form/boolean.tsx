import Checkbox from '@material-ui/core/Checkbox'
import FormControl from '@material-ui/core/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import { getMapToFromProps, IFieldProps, useFieldWithValidation } from "@xpfw/form"
import { get } from "lodash"
import { observer } from "mobx-react-lite"
import * as React from "react"
import getLabel from "./getLabel"
import { InputLabel } from '@material-ui/core'

const changeWrapper = (fieldHelper: {setValue: Function, value: any}) => {
  return (e: any) => {
    fieldHelper.setValue(!fieldHelper.value)
  }
}

const BooleanField: React.FunctionComponent<IFieldProps & {
  className?: string
  placeholder?: string
}> = observer((props) => {
  const mapTo = getMapToFromProps(props)
  const fieldHelper = useFieldWithValidation(props.schema, mapTo, props.prefix)
  const setBool = changeWrapper(fieldHelper)
  return (
    <div className="flex horizontal inputMargin">
      <FormControlLabel
          control={
            <>
            <Checkbox
              checked={fieldHelper.value}
              onChange={setBool}
              value={fieldHelper.value}
            />
            </>
          }
          label={getLabel(mapTo)}
      />
      <FormHelperText className="center">{get(props, "schema.description")}</FormHelperText>
    </div>
  )
})

export default BooleanField
export {
  changeWrapper
}