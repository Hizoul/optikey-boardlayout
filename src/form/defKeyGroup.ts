import { ExtendedJSONSchema } from "@xpfw/form"
import { backgroundColorSchema } from "./def"

const keyDisabledBackgroundSchema: ExtendedJSONSchema = {
  title: "KeyDisabledBackground", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const keyDisabledForegroundSchema: ExtendedJSONSchema = {
  title: "KeyDisabledForeground", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const keyDisabledOpacitySchema: ExtendedJSONSchema = {
  title: "KeyDisabledOpacity", type: "number"
}

const keyDownBackgroundSchema: ExtendedJSONSchema = {
  title: "KeyDownBackground", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const keyDownForegroundSchema: ExtendedJSONSchema = {
  title: "KeyDownForeground", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const keyDownOpacitySchema: ExtendedJSONSchema = {
  title: "KeyDownOpacity", type: "number"
}

const foregroundColorSchema: ExtendedJSONSchema = {
  title: "ForegroundColor", type: "string", theme: "color",
  description: "for valid colors refer to https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color"
}

const opacitySchema: ExtendedJSONSchema = {
  title: "Opacity", type: "number"
}

const autoScaleWidthSchema: ExtendedJSONSchema = {
  title: "AutoScaleToOneKeyWidth", type: "boolean",
  description: "only required to modify scaling behavior when key width is greater than 1. if not declared it will default to true"
}

const autoScaleHeightSchema: ExtendedJSONSchema = {
  title: "AutoScaleToOneKeyHeight", type: "boolean",
  description: "only required to modify scaling behavior when key width is greater than 1. if not declared it will default to true"
}
const sharedSizeSchema: ExtendedJSONSchema = {
  title: "SharedSizeGroup", type: "boolean",
  description: "only required to break out a key, or set of keys, to size separately, otherwise size grouping is determined automatically"
}

const usePersianSchema: ExtendedJSONSchema = {
  title: "UsePersianCompatibilityFont", type: "boolean"
}

const useUnicodeSchema: ExtendedJSONSchema = {
  title: "UseUnicodeCompatibilityFont", type: "boolean"
}

const useUrduSchema: ExtendedJSONSchema = {
  title: "UseUrduCompatibilityFont", type: "boolean"
}

const lockOnSchema: ExtendedJSONSchema = {
  title: "LockOnTime", type: "number",
  description: "milliseconds from when gaze is detected on key to the start of the progress animation"
}

const completionTimeSchema: ExtendedJSONSchema = {
  title: "CompletionTime", type: "number",
  description: "milliseconds from the start of the progress animation to the key being triggered"
}
const repeatDelaySchema: ExtendedJSONSchema = {
  title: "RepeatDelay", type: "number",
  description: "milleseconds between the initial keystroke and a second consecutive keystroke of the same key"
}
const repeatRateSchema: ExtendedJSONSchema = {
  title: "RepeatRate", type: "number",
  description: "milliseconds between additional consecutive keystrokes of the same keymilliseconds from when gaze is detected on key to the start of the progress animation"
}


const keyGroupSchema: ExtendedJSONSchema = {
  type: "object",
  title: "KeyGroup",
  properties: {
    [String(backgroundColorSchema.title)]: backgroundColorSchema,
    [String(foregroundColorSchema.title)]: foregroundColorSchema,
    [String(keyDisabledBackgroundSchema.title)]: keyDisabledBackgroundSchema,
    [String(keyDisabledForegroundSchema.title)]: keyDisabledForegroundSchema,
    [String(keyDisabledOpacitySchema.title)]: keyDisabledOpacitySchema,
    [String(keyDownBackgroundSchema.title)]: keyDownBackgroundSchema,
    [String(keyDownForegroundSchema.title)]: keyDownForegroundSchema,
    [String(keyDownOpacitySchema.title)]: keyDownOpacitySchema,
    [String(opacitySchema.title)]: opacitySchema,
    [String(autoScaleWidthSchema.title)]: autoScaleWidthSchema,
    [String(autoScaleHeightSchema.title)]: autoScaleHeightSchema,
    [String(sharedSizeSchema.title)]: sharedSizeSchema,
    [String(usePersianSchema.title)]: usePersianSchema,
    [String(useUnicodeSchema.title)]: useUnicodeSchema,
    [String(useUrduSchema.title)]: useUrduSchema,
    [String(lockOnSchema.title)]: lockOnSchema,
    [String(completionTimeSchema.title)]: completionTimeSchema,
    [String(repeatDelaySchema.title)]: repeatDelaySchema,
    [String(repeatRateSchema.title)]: repeatRateSchema
  }
}

export {
  keyGroupSchema
}