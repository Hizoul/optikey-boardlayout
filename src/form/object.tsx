import { getMapToFromProps, IFieldProps, SharedField, useObject, FormStore } from "@xpfw/form"
import { observer } from "mobx-react-lite"
import * as React from "react"
import actionKeyList from "../util/actionKeys"
import { keyActionObjectSchema, actionNames} from "./def"

const ObjectField: React.FunctionComponent<IFieldProps> = observer((props) => {
  const mapTo = getMapToFromProps(props)
  const fieldHelper = useObject(props.schema, mapTo, props.prefix)
  let subFieldContent
  if (props.schema != null && props.schema.title != null && props.schema.title.indexOf("keyActionObject") !== -1) {
    const currentSubType = FormStore.getValue(mapTo, props.prefix)
    if (currentSubType != null) {
      if (actionNames.indexOf(currentSubType.actionType) !== -1) {
        let type = "string"
        let theme = "default"
        let selectOptions: any = null
        if (currentSubType.actionType == "Wait") {
          type = "number"
        } else if (currentSubType.actionType == "Action") {
          theme = "select"
          selectOptions = actionKeyList.map((v) => ({label: v, value: v}))
        }
        subFieldContent = <SharedField schema={{type, title: "value", theme, selectOptions}} mapTo={`${mapTo}.value`} prefix={props.prefix} />
        if (currentSubType.actionType == "ChangeKeyboard") {
          subFieldContent = (
            <>
              {subFieldContent}
              <SharedField schema={{type: "boolean", title: "BackReturnsHere"}} mapTo={`${mapTo}.BackReturnsHere`} prefix={props.prefix} />
            </>
          )
        }
        if (currentSubType.actionType == "Loop") {
          subFieldContent = (
            <>
              <SharedField schema={{type: "number", title: "Wait"}} mapTo={`${mapTo}.Wait`} prefix={props.prefix} />
              <SharedField schema={{type: "number", title: "Count"}} mapTo={`${mapTo}.Count`} prefix={props.prefix} />
              <SharedField schema={{type: "array", items: keyActionObjectSchema, title: "keyActions"}} mapTo={`${mapTo}.keyActions`} prefix={props.prefix} />
            </>
          )
        }
      }
    }
  }  
  return (
    <div className="flex1 vertical">
      {fieldHelper.fields.map((subField) => <SharedField {...subField} key={subField.mapTo} />)}
      {subFieldContent}
    </div>
  )
})

export default ObjectField
