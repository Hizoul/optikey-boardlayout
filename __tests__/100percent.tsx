import * as React from "react"
import { changeWrapper } from "../src/form/boolean"
import SelectField, { changeWrapperSelect } from "../src/form/select"
import renderSnapshot from "../src/testUtil/renderSnapshot"
import { Transition } from "../src/rightPanel/about"
import { FormStore } from "@xpfw/form"
import { toJS } from "@xpfw/data"
import {isEqual} from "lodash"

test("100percent", () => {
  // test boolean change function
  const fieldHelper: any = {value: undefined}
  fieldHelper.setValue = (newVal: any) => fieldHelper.value = newVal
  let changer = changeWrapper(fieldHelper)
  changer(null)
  expect(fieldHelper.value).toBe(true)
  changer(null)
  expect(fieldHelper.value).toBe(false)

  const changerSelect = changeWrapperSelect(fieldHelper, "I.bla", "blu")
  changerSelect({nativeEvent: {target: {getAttribute: () => "customVal"}}})
  expect(fieldHelper.value === "customVal").toBeTruthy()
  FormStore.setValue("blu", {I: {bla: "VAL"}})
  console.log("GETTING", toJS(FormStore.getValue("blu")), {bla: "VAL"}, isEqual(toJS(FormStore.getValue("blu")), {bla: "VAL"}))
  expect(isEqual(toJS(FormStore.getValue("blu")), {I: {bla: "VAL"}})).toBe(true)
  changerSelect({nativeEvent: {target: {getAttribute: () => null}}})
  console.log("GETTING", toJS(FormStore.getValue("blu")), {bla: "VAL"}, isEqual(toJS(FormStore.getValue("blu")), {bla: "VAL"}))
  expect(isEqual(toJS(FormStore.getValue("blu")), {I: {}})).toBe(true)
  const selectSchema: any = {
    title: "my select with func",
    selectOptions: () => ["my ", "select", "function", "args"]
  }
  renderSnapshot(<SelectField schema={selectSchema} />, "select field with function args")

  renderSnapshot(<Transition/>, "transition render")
})