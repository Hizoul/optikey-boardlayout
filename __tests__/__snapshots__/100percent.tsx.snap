// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`100percent: select field with function args 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
  >
    my select with func
  </label>
  <select>
    <option />
    <option />
    <option />
    <option />
  </select>
</div>
`;

exports[`100percent: transition render 1`] = `
<div>
  slide 
  {
  "direction": "up"
}
</div>
`;
